src/kvstore/DiskManager.cpp

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+
+#include "kvstore/DiskManager.h"
+
+DEFINE_int32(disk_check_interval_secs, 10, "interval to check free space of data path");
+DEFINE_uint64(minimum_reserved_bytes, 1e9, "minimum reserved bytes of each data path");
+
+namespace nebula {
+namespace kvstore {
+
+DiskManager::DiskManager(const std::vector<std::string>& dataPaths,
+                         std::shared_ptr<thread::GenericWorker> bgThread)
+    : bgThread_(bgThread) {
+    try {
+        // atomic is not copy-constructible
+        std::vector<std::atomic_uint64_t> freeBytes(dataPaths.size() + 1);
+        // add a dummy path to indicate invalid part
+        dataPaths_.emplace_back("");
```

Comment:
What is the purpose of this? Still don't quite understand.

Reply:
In fact, it doesn't has much effect. The trick part is in `hasEnoughSpace`, all invalid part will get a index = 0, which points to the dummy path, and will return false.

I will remove the dummy path later.